modules/kafka-cluster/main.tf

```diff
@@ -10,16 +11,27 @@ resource "confluent_kafka_cluster" "cluster" {
   region       = var.region
 
   dynamic "basic" {
-    for_each = local.build_cluster_type_basic
+    for_each = var.cluster_type == "basic" ? [1] : []
     content {}
   }
 
   dynamic "standard" {
-    for_each = local.build_cluster_type_standard
+    for_each = var.cluster_type == "standard" ? [1] : []
     content {}
   }
-
+  dynamic "dedicated" {
+    for_each = var.cluster_type == "dedicated" ? [1] : []
+    content {
+      cku = var.cku
+    }
+  }
   environment {
     id = var.environment_id
   }
+  dynamic "network" {
+    for_each = var.cluster_type == "dedicated" ? [1] : []
+    content {
```

Comment:
```suggestion
    for_each = var.cluster_type == "dedicated" ? [1] : []

    content {
```

Reply:
I don't see these or the other ones above done?

Reply:
I have done this for all depending upon the cluster-type  variable. Also for dedicated type cluster only we need network id.